src/controllers/wallet/wallet.controller.js

```diff
@@ -239,21 +253,30 @@ async function getTxConfirmationBlocks(req, res) {
   res.send({ success: true, confirmationNumber });
 }
 
+/**
+ * @param {Request} req
+ * @param {Response} res
+ */
 async function simpleSendTx(req, res) {
   const validationResult = parametersValidation(req);
   if (!validationResult.success) {
     res.status(400).json(validationResult);
     return;
   }
 
-  const canStart = lock.get(req.walletId).lock(lockTypes.SEND_TX);
+  const walletLock = lock.get(req.walletId);
+  const canStart = walletLock.lock(lockTypes.SEND_TX);
   if (!canStart) {
     res.send({ success: false, error: cantSendTxErrorMessage });
     return;
   }
-  /**
-   * @type {{wallet: HathorWallet, logger: import('winston').Logger}}
-   */
+  let lockReleased = false;
+  const unlock = () => {
+    if (!lockReleased) {
+      walletLock.unlock(lockTypes.SEND_TX);
+    }
+    lockReleased = true;
+  };
```

Comment:
This `unlock` function was made to avoid calling `walletLock.unlock` twice since we call unlock after the `prepare-tx` phase and also call in the finally.

The unlock in the `finally` is required because if any errors happens before the `runSendTransaction` is called we need to unlock the send-tx lock.

Since both `unlock`s are required and calling it for a second time may cause issues we need this method that can only be called once.